nomad/fsm.go

```diff
@@ -663,7 +686,10 @@ func (n *nomadFSM) applyUpdateEval(buf []byte, index uint64) interface{} {
 }
 
 func (n *nomadFSM) upsertEvals(index uint64, evals []*structs.Evaluation) error {
-	if err := n.state.UpsertEvals(index, evals); err != nil {
+	if err := n.state.UpsertEvals(index, evals); len(evals) == 1 && err == state.ErrDuplicateEval {
+		// the request is a duplicate, ignore processing it
```

Comment:
Update the comment to explain _why_ this can happen, not just what the code is checking. The code is fairly self-explanatory on its own.

Reply:
(or not if we can remove this check altogether by gating the atomic job+eval form on server-min-ver as per https://github.com/hashicorp/nomad/pull/8435#discussion_r454606276 )

Reply:
I'm inlined to remove it then.  `state.ErrDuplicateEval` already has documentation for that, and I worry that sprinkling the ~same comments in many different places get higher risk of going out of sync and confusing readers even more.